### ISSUE TITLE
Load all files on the fly

### ISSUE THREAD

**Issue by ghahramani** [johnpapa/angular-styleguide#636]
Hi again :)
I want to hear your ideas about this method and its pros and cons:

What if we load html,controller,directive and etc... of each feature on the fly with resolve functionality in route file?

so with this we don't need to put a lot of scripts in index file and we can manage all of them just with route files.On the other hand we can merge all route files together and load the one file with automation tools.

Please let me know if I didn't explain clearly?

Thanks

**Comment by MarcLoupias** [comment]
TLDR : You wanna do lazy downloading of your ressources ?

I have done that on a web mobile project with requireJs to maximize the bootstrap time of the webapp. It is really a big big big pain to maintain (you have to maintain a json file to describe all your ressources and their links between them).

The bootstrap time is better with lazy downloading, so for a big webapp relying heavily on user trafic to make money it can worth it because users usually leave website if it is not loaded within a few seconds and a 3 sec gain in bootstrap can represents loads of money. 

Instead of that it is completly useless.

**Comment by johnpapa** [comment]
you can use require, browserify, oc lazy load. sure.

but i do not recommend ever having a long list of script tags in your index.html. whatever you choose, any solution that loads either a bundle or set of bundles is optimal
